shared/js/background/amp-protection.js

```diff
@@ -49,7 +49,11 @@ function extractAMPURL (site, url) {
     for (const regexPattern of ampSettings.linkFormats) {
         const match = url.match(regexPattern)
         if (match && match.length > 1) {
-            const newSite = new Site(match[1].startsWith('http') ? match[1] : `https://${match[1]}`)
+            const newUrl = match[1]
+            if (newUrl.match(/^.+:\/\//) && !newUrl.startsWith('http')) {
```

Comment:
Can you pass to the URL constructor and check scheme (also catching the potential throw for relative paths)